fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/executable/NumericArithmetic.java

```diff
@@ -596,4 +650,515 @@ public static Expression divideDecimalV3(DecimalV3Literal first, DecimalV3Litera
+        if (second.getValue() >= first.getValue().scale()) {
+            return first;
+        }
+        return first.roundFloor(second.getValue());
+    }
+
+    /**
+     * floor
+     */
+    @ExecFunction(name = "floor")
+    public static Expression floor(DoubleLiteral first) {
+        DecimalV3Literal middleResult = new DecimalV3Literal(new BigDecimal(Double.toString(first.getValue())));
+        return new DoubleLiteral(middleResult.roundFloor(0).getDouble());
+    }
+
+    /**
+     * floor
+     */
+    @ExecFunction(name = "floor")
+    public static Expression floor(DoubleLiteral first, IntegerLiteral second) {
+        DecimalV3Literal middleResult = new DecimalV3Literal(new BigDecimal(Double.toString(first.getValue())));
+        if (second.getValue() >= middleResult.getValue().scale()) {
+            return first;
+        }
+        return new DoubleLiteral(middleResult.roundFloor(second.getValue()).getDouble());
+    }
+
+    /**
+     * exp
+     */
+    @ExecFunction(name = "exp")
+    public static Expression exp(DoubleLiteral first) {
+        return new DoubleLiteral(Math.exp(first.getValue()));
+    }
+
+    /**
+     * ln
+     */
+    @ExecFunction(name = "ln")
+    public static Expression ln(DoubleLiteral first) {
+        if (first.getValue() <= 0) {
+            return new NullLiteral(first.getDataType());
+        }
+        return new DoubleLiteral(Math.log(first.getValue()));
+    }
+
+    /**
+     * log
+     */
+    @ExecFunction(name = "log")
+    public static Expression log(DoubleLiteral first, DoubleLiteral second) {
+        if (first.getValue() <= 0) {
+            return new NullLiteral(first.getDataType());
+        }
+        return new DoubleLiteral(Math.log(first.getValue()) / Math.log(second.getValue()));
+    }
+
+    /**
+     * log2
+     */
+    @ExecFunction(name = "log2")
+    public static Expression log2(DoubleLiteral first) {
+        if (first.getValue() <= 0) {
+            return new NullLiteral(first.getDataType());
+        }
+        return new DoubleLiteral(Math.log(first.getValue()) / Math.log(2.0));
+    }
+
+    /**
+     * log10
+     */
+    @ExecFunction(name = "log10")
+    public static Expression log10(DoubleLiteral first) {
+        if (first.getValue() <= 0) {
+            return new NullLiteral(first.getDataType());
+        }
+        return new DoubleLiteral(Math.log10(first.getValue()));
+    }
+
+    /**
+     * sqrt
+     */
+    @ExecFunction(name = "sqrt")
+    public static Expression sqrt(DoubleLiteral first) {
+        if (first.getValue() < 0) {
+            return new NullLiteral(first.getDataType());
+        }
+        return new DoubleLiteral(Math.sqrt(first.getValue()));
+    }
+
+    /**
+     * power
+     */
+    @ExecFunction(name = "power")
+    public static Expression power(DoubleLiteral first, DoubleLiteral second) {
+        return new DoubleLiteral(Math.pow(first.getValue(), second.getValue()));
```

Comment:
pow could return infinity or NaN in many cases, u should process them correctly: https://docs.oracle.com/javase/8/docs/api/java/lang/Math.html#pow-double-double-

please check all function one by one according to java doc

Reply:
done